src/Ocelot.Provider.Polly/OcelotBuilderExtensions.cs

```diff
@@ -1,18 +1,32 @@
 ﻿using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
+
 using Ocelot.Configuration;
 using Ocelot.DependencyInjection;
 using Ocelot.Errors;
 using Ocelot.Logging;
 using Ocelot.Provider.Polly.Interfaces;
 using Ocelot.Requester;
+
 using Polly.CircuitBreaker;
 using Polly.Timeout;
 
 namespace Ocelot.Provider.Polly;
 
 public static class OcelotBuilderExtensions
 {
+    public static IOcelotBuilder AddPolly<T>(this IOcelotBuilder builder,
+        QosDelegatingHandlerDelegate delegatingHandler,
+        IDictionary<Type, Func<Exception, Error>> errorMapping) where T : class, IPollyQoSProvider<HttpResponseMessage>
```

Comment:
@RaynaldM The issue is that - DI - ```Dictionary<Type, Func<Exception, Error>>``` is not the same as ```IDictionary<Type, Func<Exception, Error>>```. In ```Ocelot.Requester.HttpExceptionToErrorMapper``` ocelot is trying to get a service by calling 
```c#
_mappers = serviceProvider.GetService<Dictionary<Type, Func<Exception, Error>>>();
```
... It will never work as expected. I agree with you, IDictionary is the correct option, but then you will need to modify ```HttpExceptionToErrorMapper``` and fix a test case in ```Ocelot.UnitTests.Requester``` in ```Should_return_error_from_mapper```, casting the Dictionary to IDictionary, 
```c# 
_services.AddSingleton((IDictionary<Type, Func<Exception, Error>>) errorMapping);
```

Reply:
Why `IDictionary<TKey, TValue>`? Do we use methods & props of this interface?
Is it OK to use most abstract interfaces like simple `ICollection<T>` or even `IEnumerable<T>`?

Reply:
back to Dictionary